### PR TITLE
Reduce overhead in RetryStateImpl by fixing alignment

### PR DESCRIPTION
**Pull request by DiazAlan**
Commit Message: Reduce overhead in RetryStateImpl by fixing alignment
Risk Level: Low

Sizeof test
Before changes: 280
After changes: 264

### CODE REVIEW

**Comment by DiazAlan** [comment]
@adisuissa @yurykats

**Review by adisuissa — APPROVED** [review]
Thanks, left a small nit about the comment, but LGTM.

/assign-from @envoyproxy/senior-maintainers

File: source/common/router/retry_state_impl.h [hunk 1]
```diff
@@ -122,21 +122,23 @@ class RetryStateImpl : public RetryState {
...
   BackOffStrategyPtr ratelimited_backoff_strategy_{};
   std::vector<Upstream::RetryHostPredicateSharedPtr> retry_host_predicates_;
   Upstream::RetryPrioritySharedPtr retry_priority_;
-  uint32_t host_selection_max_attempts_;
   std::vector<uint32_t> retriable_status_codes_;
   std::vector<Http::HeaderMatcherSharedPtr> retriable_headers_;
   std::vector<ResetHeaderParserSharedPtr> reset_headers_{};
   std::chrono::milliseconds reset_max_interval_{};
+
+  // Keep small members (bools and enums) at the end of class, to reduce alignment overhead.
```
[the remark this answers is not in the document]
Comment (DiazAlan) [reply source/common/router/retry_state_impl.h]: Sure that's fair. Done

**Review by adisuissa — APPROVED** [review]
LGTM, thanks!

/assign-from @envoyproxy/senior-maintainers

**Review by htuch — APPROVED** [review]
LGTM, thanks!

**Comment by htuch** [comment]
Same comment as https://github.com/envoyproxy/envoy/pull/26128#issuecomment-1475573864 @DiazAlan - I think it would be helpful to have a quantitative method to track how these improvements are stacking up.

**Pull request merged**